flow/testing/mock_layer.h

```diff
@@ -46,25 +43,74 @@ class MockLayer : public Layer {
   const MutatorsStack& parent_mutators() { return parent_mutators_; }
   const SkMatrix& parent_matrix() { return parent_matrix_; }
   const SkRect& parent_cull_rect() { return parent_cull_rect_; }
-  bool parent_has_platform_view() { return parent_has_platform_view_; }
-  bool parent_has_texture_layer() { return parent_has_texture_layer_; }
 
   bool IsReplacing(DiffContext* context, const Layer* layer) const override;
   void Diff(DiffContext* context, const Layer* old_layer) override;
   const MockLayer* as_mock_layer() const override { return this; }
 
+  bool parent_has_platform_view() {
+    return mock_flags_ & kParentHasPlatformView;
+  }
+
+  bool parent_has_texture_layer() {
+    return mock_flags_ & kParentHasTextureLayer;
+  }
+
+  bool fake_has_platform_view() { return mock_flags_ & kFakeHasPlatformView; }
+
+  bool fake_reads_surface() { return mock_flags_ & kFakeReadsSurface; }
+
+  bool fake_opacity_compatible() {
+    return mock_flags_ & kFakeOpacityCompatible;
+  }
+
+  bool fake_has_texture_layer() { return mock_flags_ & kFakeHasTextureLayer; }
+
+  void set_parent_has_platform_view(bool flag) {
```

Comment:
A suggestion here would be to have these return a MockLayer& and all return *this so that you can do `auto layer = MockLayer(...).set_flag1().set_flag2();`, or, for readability:

```
  auto layer = MockLayer(...)
      .set_flag1(...)
      .set_flag2(...);
```